File: 1552-build-an-array-with-stack-operations/build-an-array-with-stack-operations.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        var ans = new ArrayList<String>();
        var st = new Stack<Integer>();
        int ptr = 0;
        for(int i=1; i<=n; i++) {
            if(ptr == target.length) break;
            else if(target[ptr] == i) {
                ans.add("Push");
                st.push(i);
                ptr++;
            }
            else {
                ans.add("Push");
                ans.add("Pop");
            }
        }
        return ans;
    }
}

File: 1552-build-an-array-with-stack-operations/build-an-array-with-stack-operations.js
/**
 * @param {number[]} target
 * @param {number} n
 * @return {string[]}
 */
var buildArray = function(target, n) {
    const ans = [];
    let ptr = 0;

    for(let i=1; i<=n; i++) {
        if(ptr === target.length) break;
        else if(target[ptr] === i) {
            ans.push("Push");
            ptr++;
        }
        else {
            ans.push("Push");
            ans.push("Pop");
        }
    }

    return ans;
};